src/main/cpp/option_processor.cc

```diff
@@ -185,19 +185,21 @@ std::set<std::string> GetOldRcPaths(
         internal::FindRcAlongsideBinary(cwd, path_to_binary);
     candidate_bazelrc_paths = {workspace_rc, binary_rc, system_bazelrc_path};
   }
-  const std::vector<std::string> deduped_blazerc_paths =
-      internal::DedupeBlazercPaths(candidate_bazelrc_paths);
-  std::set<std::string> old_rc_paths(deduped_blazerc_paths.begin(),
-                                     deduped_blazerc_paths.end());
   string user_bazelrc_path = internal::FindLegacyUserBazelrc(
       SearchUnaryOption(startup_args, "--bazelrc"), workspace);
   if (!user_bazelrc_path.empty()) {
-    old_rc_paths.insert(user_bazelrc_path);
+    candidate_bazelrc_paths.push_back(user_bazelrc_path);
   }
-  return old_rc_paths;
+  const std::vector<std::string> deduped_blazerc_paths =
+      internal::GetExistingDedupedBazelrcPaths(candidate_bazelrc_paths);
+  return std::set<std::string>(deduped_blazerc_paths.begin(),
+                               deduped_blazerc_paths.end());
 }
 
-std::vector<std::string> DedupeBlazercPaths(
+// Deduplicates the given paths based on their canonical form.
+// Computes the canonical form using blaze_util::MakeCanonical.
+// Returns the unique paths in their original form (not the canonical one).
+std::vector<std::string> GetExistingDedupedBazelrcPaths(
```

Comment:
nit: DedupeBazelRcPaths? (it's not a getter)

Also, why "Existing"? It doesn't check existence, does it?

Reply:
The canonical path can only be computed of existing paths, so the paths indeed exist.

Renamed the method to its old name.